### PR TITLE
Run Sourcekitd-tests in a non dispatch thread

### PR DESCRIPTION
**Pull request by gmittert**
Since Dispatch threads have a 64k stack size, tests were failing due to
blowing the stack on Windows. This runs the tests in thread with a
larger stack size. This fixes some 90ish sourcekit tests on Windows.

### CODE REVIEW

**Comment by gmittert** [comment]
@compnerd

**Comment by compnerd** [comment]
CC: @akyrtzi @benlangmuir

**Comment by compnerd** [comment]
@swift-ci please test

File: tools/SourceKit/lib/SwiftLang/SwiftASTManager.cpp [hunk 1]
```diff
@@ -197,7 +197,7 @@ namespace SourceKit {
         LOG_WARN_FUNC("did not find primary SourceFile");
         ConsumerRef->failed("did not find primary SourceFile");
       }
-    });
+    }, /*isStackDeep=*/true);
```
Comment (akyrtzi) [inline tools/SourceKit/lib/SwiftLang/SwiftASTManager.cpp]: This is unfortunate, it's going to be creating a new pthread every time to query the AST. Is there no way to make it work without making this change ?
I find strange that dispatch threads have limited stack also on Darwin but we did not have issues with querying the AST.

If there is no way around it I'd at least suggest to only have this change in effect when compiling for Windows.

**Review by akyrtzi — CHANGES_REQUESTED** [review]

File: tools/SourceKit/lib/SwiftLang/SwiftASTManager.cpp (hunk 1 above)
Comment (compnerd) [reply tools/SourceKit/lib/SwiftLang/SwiftASTManager.cpp]: https://github.com/apple/swift-corelibs-libdispatch/blob/6c5b3ba86fce0480f702ba13fcb5f656af090b20/src/queue.c#L5175

This is hardcoded into libdispatch.  There is no way to adjust the stack size.  It is very likely that you are skimming past pretty close to the limit.

Comment (gmittert) [reply tools/SourceKit/lib/SwiftLang/SwiftASTManager.cpp]: That's my current theory. A [sample count of frame sizes](https://gist.github.com/gmittert/1d6691ba799da7ce923f0c7c75130b1c) seems to confirm that. Why Windows uses slightly more memory, I'm not sure yet.

Comment (akyrtzi) [reply tools/SourceKit/lib/SwiftLang/SwiftASTManager.cpp]: That stack trace looks to be from syntactic parsing, it is unrelated to `ASTUnit::Implementation::consumeAsync`.
Is the change inside `consumeAsync` really necessary or are the other changes making it redundant ?

Comment (gmittert) [reply tools/SourceKit/lib/SwiftLang/SwiftASTManager.cpp]: The consume async change is needed here. `"S:\b\swift\bin\sourcekitd-test.EXE" "-req=cursor" "-pos=9:25" "S:\swift\test\SourceKit\CursorInfo\rdar_30248264.swift" "--" "S:\swift\test\SourceKit\CursorInfo\rdar_30248264.swift"` stack overflows without it on windows. I'm currently trying to find a better way to do it. At worst, I'll `#if defined(_WIN32)` it, but I'd rather do something nicer.

The stack trace was more an example from another test that was stack overflowing showing that Windows typically does currently use more stack memory than Darwin.

**Comment by gmittert** [comment]
Bleh, no idea what's causing the difference in stack sizes. I sadly just put it behind and if defined WIN32.

@swift-ci please test

**Review by akyrtzi — APPROVED** [review]

**Pull request merged**